doc/style-guide-javascript.md

```diff
@@ -0,0 +1,185 @@
+Style Guide for JavaScript
+==========================
+
+This extends the [general programming](style-guide-programming.md) portion of the [Style Guide](style-guide.md).
+
+
+* Filenames
+    * Tests for `*.js` files are named `*.spec.js`.
+    * Tests live either with the JavaScript being developed or within a `spec/` folder.  If the folder exists, all tests must be within it.
+    * Angular directives, services, etc. have their type after the name (eg `some-directive-service.js` and `datepicker-directive.js`).
+    * Related files should be named similarly (eg `response-template-service.js`, `response-template-service.css`, `response-template-service-declined.html`, `response-template-service-approved.html`).
+* Capitalization and Naming
+    * Variables and methods should use `camelCase` style formatting.
+    * Only constructors should use `UpperCamelCase` formatting, such as `Date` and `Promise`.
+    * No underscore before, within, nor after variables, functions, or properties.
+    * Functions shall be named (eg. `function MyFunc()`) only if they are not ever supposed to be overridden.  Anonymous functions don't need a name.
```

Comment:
I am a little confused here. Are you saying that functions that are not ever supposed to be overridden should be `UpperCamelCase`? The wording on this particular line threw me off a little bit but I am probably just slow.